From: Pedro Falcato <pfalcato@suse.de>
Date: Mon, 2 Jun 2025 13:02:28 +0100
Subject: kABI: ipv6: save dontfrag in cork
Patch-mainline: Never, kABI workaround
References: git-fixes

The patch "patches.suse/ipv6-save-dontfrag-in-cork.patch" breaks kABI by adding
a new field to inet6_cork (used in ipv6_pinfo). Hide it from genksyms. As the
patch suggests, there's a 6 byte field at the end of inet6_cork (due to
alignment reasons), so the patch is safe.

Signed-off-by: Pedro Falcato <pfalcato@suse.de>
[pfalcato: Ported to SLE15-SP6]
---
 include/linux/ipv6.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/ipv6.h
+++ b/include/linux/ipv6.h
@@ -207,7 +207,9 @@ struct inet6_cork {
 	struct ipv6_txoptions *opt;
 	u8 hop_limit;
 	u8 tclass;
+#ifndef __GENKSYMS__
 	u8 dontfrag:1;
+#endif
 };
 
 /* struct ipv6_pinfo - ipv6 private area */
